shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,237 @@
+                [("width", "=", 0)],
+                [("width", "=", False)],
+                [("height", "=", 0)],
+                [("height", "=", False)],
+                [("max_weight", "=", 0)],
+                [("max_weight", "=", False)],
+                [("qty", "=", 0)],
+                [("qty", "=", False)],
+                [("barcode", "=", False)],
+            ]
+        )
+
+    def _get_next_packaging_to_set_dimension(self, product, previous_packaging=None):
+        """Return for a product the next packaging needing dimension to be set."""
+        next_packaging_id = previous_packaging.id + 1 if previous_packaging else 0
+        domain_dimension = self._get_domain_packaging_needs_dimension()
+        domain_packaging_id = [
+            ("product_id", "=", product.id),
+            ("id", ">=", next_packaging_id),
+        ]
+        domain = expression.AND([domain_packaging_id, domain_dimension])
+        return self.env["product.packaging"].search(domain, order="id", limit=1)
+
+    def _response_for_set_packaging_dimension(
+        self, picking, line, packaging=None, message=None
+    ):
+        if not packaging:
+            packaging = self._get_next_packaging_to_set_dimension(line.product_id)
+        if not packaging:
+            return self._response_for_set_quantity(picking, line, message=message)
+        return self._response(
+            next_state="set_packaging_dimension",
+            data={
+                "picking": self.data.picking(picking),
+                "selected_move_line": self.data.move_line(line),
+                "packaging": self.data_detail.packaging_detail(packaging),
+            },
+            message=message,
+        )
+
+    def _store_dimension_to_update(self, dimensions, local_table):
+        """Store values to update in the context manager of the component.
+
+        It helps for extending the module to add more dimension fields to be updated.
+
+        """
+        for dimension in dimensions:
+            if local_table[dimension] is None:
+                if dimension in self.dimension_to_update.keys():
+                    self.dimension_to_update.pop(dimension)
+            else:
+                self.dimension_to_update[dimension] = local_table[dimension]
+
+    def set_packaging_dimension(
+        self,
+        picking_id,
+        selected_line_id,
+        packaging_id,
+        height=None,
+        length=None,
+        width=None,
+        weight=None,
+        qty=None,
+        barcode=None,
+        cancel=False,
+    ):
+        """Set the dimension on a product packaging.
+
+        If the user cancel the dimension update we still propose the next
+        possible packgaging.
+
+        Transitions:
+            - set_packaging_dimension: if more packaging needs dimension
+            - set_quantity: otherwise
+        """
+        self._store_dimension_to_update(
+            ["height", "length", "width", "weight", "barcode", "qty"], locals()
```

Comment:
I would not pass `locals`. Can't we wrap dimension in its own dict? IMO we should have `measures` dict containing all of them.

Reply:
Why don't you use **kwargs?
Then the store_dimension_to_update is not needed
`def set_packaging_dimension(self, picking_id, selected_line_id, packaging_id, cancel=False, **kwargs):`

Reply:
You are right, it seems the solution was over complex. Now no need to use the work context of the component or the `locals`.
I refactored.